cfme/storage/volume.py

```diff
@@ -243,6 +287,36 @@ def create_snapshot(self, name, cancel=False, reset=False):
                 view.save.click()
                 return snapshot_collection.instantiate(name, self.provider)
 
+    def attach_instance(self, name, mountpoint=None, cancel=False, reset=False):
+        view = navigate_to(self, 'AttachInstance')
+
+        # Reset and Attach buttons are only active when view is changed
+        changed = view.fill({'instance': name, 'mountpoint': mountpoint})
+
+        if cancel:
+            view.cancel.click()
+
+        elif changed:
+            if reset:
+                view.reset.click()
+            else:
+                view.attach.click()
+
+        view.flash.assert_no_error()
```

Comment:
@jadhaj Was there a reason this did not create a view? I believe this would be beneficial, then you would not have to create the view everytime in your tests.

Reply:
@jawatts Sorry but i am not sure i understand what is needed here

Reply:
@jawatts I guess you mean to create a View and return it at end of method ?
The reason is that  I just did not see examples on the code for such behavior 
Usually we return instances not Views

Reply:
@jadhaj you are correct! It would be something like 

```view = self.create_view(VolumeDetailsView)```

That way when you are assert no errors it will be agaisnt the` VolumeDetailsView`